spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -411,6 +411,9 @@ public void setPublisherConfirms(boolean publisherConfirms) {
 		if (publisherConfirms) {
 			setPublisherConfirmType(ConfirmType.CORRELATED);
 		}
+		else if (this.confirmType.equals(ConfirmType.CORRELATED)) {
+			this.confirmType = ConfirmType.NONE;
```

Comment:
I think we still need to call `setPublisherConfirmType()`, since we need to propagate a state into a `this.publisherConnectionFactory` as well...

Reply:
Doh - good catch - not enough coffee 😦